hazelcast/src/main/java/com/hazelcast/config/CacheConfig.java

```diff
@@ -135,6 +143,55 @@ public CacheConfig(CacheSimpleConfig simpleConfig) throws Exception {
         }
     }
 
+    private void initExpiryPolicyFactoryConfig(CacheSimpleConfig simpleConfig) throws Exception {
+        CacheSimpleConfig.ExpiryPolicyFactoryConfig expiryPolicyFactoryConfig =
+                simpleConfig.getExpiryPolicyFactoryConfig();
+        if (expiryPolicyFactoryConfig != null) {
+            if (expiryPolicyFactoryConfig.getClassName() != null) {
+                this.expiryPolicyFactory =
+                        ClassLoaderUtil.newInstance(null, expiryPolicyFactoryConfig.getClassName());
+            } else {
+                TimedExpiryPolicyFactoryConfig timedExpiryPolicyConfig =
+                        expiryPolicyFactoryConfig.getTimedExpiryPolicyFactoryConfig();
+                if (timedExpiryPolicyConfig != null) {
+                    DurationConfig durationConfig = timedExpiryPolicyConfig.getDurationConfig();
+                    ExpiryPolicyType expiryPolicyType = timedExpiryPolicyConfig.getExpiryPolicyType();
+                    switch (expiryPolicyType) {
+                        case CREATED:
+                            this.expiryPolicyFactory =
+                                    CreatedExpiryPolicy.factoryOf(
+                                            new Duration(durationConfig.getTimeUnit(),
```

Comment:
you are creating a Duration object for each policy, how about creating it once before switch/case.
whole switch/case can be pulled out to method for simplicity